.github/workflows/aws-deployment.yml

```diff
@@ -1,38 +1,86 @@
 name: AWS Deployment
-on: [push, pull_request]
+on: [push]
 jobs:
   build:
     runs-on: ubuntu-latest
-    name: Build Todo App
+    name: Build and deploy Todo App
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/cache@v1
+
+      - name: Checkout code
+        uses: actions/checkout@v1
+
+      - name: Gradle cache
+        uses: actions/cache@v1
         with:
           path: ~/.gradle/caches
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
+
       - name: Set up JDK 11
         uses: actions/setup-java@v1
         with:
           java-version: 11.0
+
       - name: Build application
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+        run: |
+          cd application
+          ./gradlew build --stacktrace
+
+      - name: "Zip build reports"
+        if: failure()
+        run: zip -r reports.zip **/build/reports
+
+      - name: "Upload build reports"
+        uses: actions/upload-artifact@v1
+        if: failure()
+        with:
+          name: reports
+          path: reports.zip
+
+      - name: Create or update ECR registry stack
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: ${{ secrets.AWS_REGION }}
+        run: |
+          cd cloudformation/application-setup-basic/stacks/registry
+          ./create-or-update-registry-stack.sh
+
+      - name: Create or update Cognito stack
```

Comment:
Creating or updating the Cognito stack before the application stack because we need the Cognito client secret in the application stack (and CloudFormation sadly doesn't provide a way to pass the secret created by Cognito directly from one stack to the other).